abi/abi.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package abi
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/utils/set"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/consts"
+
+	reflect "reflect"
+)
+
+type VM struct {
+	Actions []Action `serialize:"true" json:"actions"`
+}
+
+var _ codec.Typed = (*VM)(nil)
+
+const ABITypeID = consts.MaxUint8
+
+func (VM) GetTypeID() uint8 {
+	return ABITypeID
+}
+
+func (a *VM) Hash() [32]byte {
+	writer := codec.NewWriter(0, consts.NetworkSizeLimit)
+	err := codec.LinearCodec.MarshalInto(a, writer.Packer)
+	if err != nil {
```

Comment:
I strongly disagree. I prefer error handling on a separate line - it looks cleaner to me. Do we have a standard for that? It would be nice to agree on one style or another.

Reply:
We use the in-line format when a single error value is returned